pkg/workloads/reservation.go

```diff
@@ -1318,6 +1295,97 @@ func (a *API) setFarmIPFree(db *mongo.Database, workload types.WorkloaderType) e
 	return nil
 }
 
+func (a *API) handleKubernetesReservation(db *mongo.Database, workload types.WorkloaderType, userID int64) mw.Response {
+	k8sWorkload := workload.Workloader.(*generated.K8S)
+
+	if k8sWorkload.PublicIP == 0 {
+		return nil
+	}
+
+	var err error
+
+	var workloadFiler types.WorkloadFilter
+	workloadFiler = workloadFiler.WithID(k8sWorkload.PublicIP)
+	ipWorkload, err := workloadFiler.Get(context.Background(), db)
+	if err != nil {
+		return mw.Error(err)
+	}
+
+	if ipWorkload.GetNextAction() != generated.NextActionDeploy {
+		customErr := fmt.Errorf("ip reservation: %d, is not valid anymore", ipWorkload.WorkloadID())
+		if err := a.updateReservationResult(db, customErr, workload); err != nil {
+			return mw.Error(err)
+		}
+		return nil
+	}
+
+	if ipWorkload.GetCustomerTid() != userID {
+		customErr := fmt.Errorf("user does not own this IP reservation: %d, for this kubernetes cluster", ipWorkload.WorkloadID())
+		if err := a.updateReservationResult(db, customErr, workload); err != nil {
+			return mw.Error(err)
+		}
+		return nil
+	}
+
+	workloadFiler = workloadFiler.WithCustomerID(int(userID))
+	workloadFiler = workloadFiler.WithNextAction(generated.NextActionDeploy)
+	workloads, err := workloadFiler.Find(context.Background(), db)
+	if err != nil {
+		return mw.Error(err)
+	}
+
+	// Try to find a kubernetes reservation that already references the public IP reservation that the user wants to provision
+	// with his new kubernetes reservation
+	for _, workload := range workloads {
```

Comment:
Please add more filters to the workloads so u can do this in the db instead of code